runtime/observability/telemetry-provider-otel/jvm/src/aws/smithy/kotlin/runtime/telemetry/otel/AttributeUtils.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.telemetry.otel
+
+import aws.smithy.kotlin.runtime.util.AttributeKey
+import aws.smithy.kotlin.runtime.util.Attributes
+import aws.smithy.kotlin.runtime.util.get
+import io.opentelemetry.api.common.AttributeKey as OtelAttributeKey
+import io.opentelemetry.api.common.Attributes as OtelAttributes
+
+@Suppress("UNCHECKED_CAST")
+internal fun Attributes.toOtelAttributes(): OtelAttributes {
+    val keys = this.keys
+    if (keys.isEmpty()) return OtelAttributes.empty()
+    val attrs = OtelAttributes.builder()
+    keys.forEach {
+        val key = it as AttributeKey<Any>
+        val value = get(key)
+        it.otelAttrKeyOrNull(value)?.let { otelKey ->
+            attrs.put(otelKey, value)
+        }
+    }
+
+    return attrs.build()
+}
+
+internal fun <T> AttributeKey<T>.otelAttrKeyOrNull(value: T): OtelAttributeKey<T>? {
+    val otelKey = when (value) {
+        is String -> OtelAttributeKey.stringKey(name)
+        is Long -> OtelAttributeKey.longKey(name)
+        is Boolean -> OtelAttributeKey.booleanKey(name)
+        is Double -> OtelAttributeKey.doubleKey(name)
+        is List<*> -> {
+            when (value.firstOrNull()) {
```

Comment:
This won't work if it's a mixed type `List<Any>`, is that ok?

Reply:
Yeah it's fine we are in control of what attributes we are setting, if we are setting `List<Any>` we are doing something wrong probably